File: WebApi/Models/Horse.cs
public class Horse : IModel, IEnityWithTreatments, IEnityWithFiles
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public int NumberOfWeeksUntilNextTreatmentHoofcare { get; set; } = 0;
    public int NumberOfWeeksUntilNextTreatmentToothcare { get; set; } = 0;
    public int BirthYear { get; set; } = 0;
    public string NoteForNextTreatment { get; set; } = string.Empty;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
    public bool Beschlagen { get; set; } = false;
    public string FileKeysString { get; set; } = string.Empty;
    public ICollection<Treatment> Treatments { get; set; } = new List<Treatment>();
    public ICollection<File> Files { get; set; } = new List<File>();
    public string Patenschaften { get; set; } = string.Empty;
}

File: vue-app/src/classes/Horse.ts
import { IHorse } from "../interfaces/IHorse";
import { LocalIDFactory } from "@/helpers/LocalIDFactory";
import { IFile } from "@/interfaces/IFile";
import { IRoutine } from "@/interfaces/IRoutine";
import { ITreatmentDate } from "@/interfaces/ITreatmentDate";

export class Horse implements IHorse {
  localID?: string;
  id?: number;
  name: string;
  numberOfWeeksUntilNextTreatmentHoofcare: number;
  numberOfWeeksUntilNextTreatmentHoofcareFollowUp1: number;
  followUp1AdvanceNotice: string;
  numberOfWeeksUntilNextTreatmentHoofcareFollowUp2: number;
  followUp2AdvanceNotice: string;
  numberOfWeeksUntilNextTreatmentToothcare: number;
  numberOfWeeksUntilNextTreatmentHealthcare: number;
  birthYear: number;
  noteForNextTreatment: string;
  createdAt: Date;
  updatedAt: Date;
  beschlagen: boolean;
  fileKeysString: string;
  files: IFile[];
  treatmentIds: number[];
  treatmentDates: ITreatmentDate[];
  fileIds: number[];
  patenschaften: string;
  summaryHoofCheckStatusOfLastTreatment: string;
  workOnHoof: string;
  includedRoutines: IRoutine[];
  routineIds: number[];
  color: string;

  constructor() {
    this.localID = LocalIDFactory.createLocalID();
    this.name = "";
    this.numberOfWeeksUntilNextTreatmentHoofcare = 8;
    this.numberOfWeeksUntilNextTreatmentHoofcareFollowUp1 = 0;
    this.followUp1AdvanceNotice = "";
    this.numberOfWeeksUntilNextTreatmentHoofcareFollowUp2 = 0;
    this.followUp2AdvanceNotice = "";
    this.numberOfWeeksUntilNextTreatmentToothcare = 8;
    this.numberOfWeeksUntilNextTreatmentHealthcare = 8;
    this.birthYear = 0;
    this.noteForNextTreatment = "";
    this.createdAt = new Date();
    this.updatedAt = new Date();
    this.beschlagen = false;
    this.fileKeysString = "";
    this.treatmentIds = [];
    this.files = [];
    this.treatmentDates = [];
    this.fileIds = [];
    this.patenschaften = "";
    this.summaryHoofCheckStatusOfLastTreatment = "";
    this.workOnHoof = "";
    this.includedRoutines = [];
    this.routineIds = [];
    this.color = "#FFFFFF";
  }
}
